src/core/permissions/sagas/getDataSetPermissionsPage.js

```diff
@@ -0,0 +1,198 @@
+/*
+ * @flow
+ */
+
+import {
+  call,
+  put,
+  select,
+  takeEvery,
+} from '@redux-saga/core/effects';
+import { List, Map, Set } from 'immutable';
+import { Models } from 'lattice';
+import {
+  AxiosUtils,
+  DataUtils,
+  LangUtils,
+  Logger,
+} from 'lattice-utils';
+import type { Saga } from '@redux-saga/core';
+import type {
+  Ace,
+  Organization,
+  PermissionType,
+  Principal,
+  PropertyType,
+  UUID,
+} from 'lattice';
+import type { WorkerResponse } from 'lattice-sagas';
+import type { SequenceAction } from 'redux-reqseq';
+
+import { getPermissionsWorker } from './getPermissions';
+
+import { ERR_MISSING_ORG, ERR_MISSING_PROPERTY_TYPE } from '../../../utils/constants/errors';
+import { getOrgDataSetColumnsFromMeta, getOrgDataSetsFromMeta } from '../../edm/actions';
+import { FQNS } from '../../edm/constants';
+import { getOrgDataSetColumnsFromMetaWorker, getOrgDataSetsFromMetaWorker } from '../../edm/sagas';
+import { PAGE_PERMISSIONS_BY_DATA_SET, TOTAL_PERMISSIONS } from '../../redux/constants';
+import {
+  selectOrgDataSets,
+  selectOrgDataSetsColumns,
+  selectOrganization,
+  selectPrincipalPermissions,
+  selectPropertyTypes,
+} from '../../redux/selectors';
+import { GET_DATA_SET_PERMISSIONS_PAGE, getDataSetPermissionsPage, getPermissions } from '../actions';
+
+const { FQN } = Models;
+const { toSagaError } = AxiosUtils;
+const { getPropertyValue } = DataUtils;
+const { isNonEmptyArray, isNonEmptyString } = LangUtils;
+
+const REQUIRED_PROPERTY_TYPES :FQN[] = [
+  FQNS.OL_DATA_SET_ID,
+];
+
+const LOG = new Logger('PermissionsSagas');
+
+function* getDataSetPermissionsPageWorker(action :SequenceAction) :Saga<void> {
+
+  try {
+    yield put(getDataSetPermissionsPage.request(action.id, action.value));
+
+    const {
+      filterByPermissionTypes,
+      filterByQuery,
+      initialize,
+      organizationId,
+      pageSize,
+      principal,
+      start,
+    } :{|
+      filterByPermissionTypes :Array<PermissionType>;
+      filterByQuery :string;
+      initialize :boolean;
+      organizationId :UUID;
+      pageSize :number;
+      principal :Principal;
+      start :number;
+    |} = action.value;
+    let response :WorkerResponse;
+
+    const organization :?Organization = yield select(selectOrganization(organizationId));
+    if (!organization) {
+      throw new Error(ERR_MISSING_ORG);
+    }
+
+    const propertyTypes :Map<UUID, PropertyType> = yield select(selectPropertyTypes(REQUIRED_PROPERTY_TYPES));
+    const propertyTypeIds :Map<FQN, UUID> = propertyTypes.map((propertyType) => propertyType.type).flip();
+    if (propertyTypeIds.count() !== REQUIRED_PROPERTY_TYPES.length) {
+      throw new Error(ERR_MISSING_PROPERTY_TYPE);
+    }
+
+    if (initialize === true) {
+      // NOTE: we only want to call getOrgDataSetsFromMeta once, not for every page
```

Comment:
why not just `if(initialize) {...}`

Reply:
Just really wanted to hammer home the fact that this is a boolean. I can change it, but I don't see the harm of keeping it.